lib/iris/fileformats/grib/_message.py

```diff
@@ -110,6 +108,8 @@ def data(self):
                 raise TranslationError(msg)
             if template == 90:
                 shape = (grid_section['Ny'], grid_section['Nx'])
+            elif template == 40 and reduced:
+                shape = (grid_section['numberOfDataPoints'],)
```

Comment:
(It also looks like the same sort of issue applies to the `grid_definition_section()` function?)